### PR TITLE
`--dev` implies `--tmp`

### PR DESCRIPTION
**Pull request by bkchr**
This changes `--dev` to imply `--tmp` when no explicit `--base-path` is provided.

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

**Review by sorpaas — APPROVED** [review]

**Comment by xlc** [comment]
this is a breaking change to developers.

**Review by KiChjang — APPROVED** [review]

**Comment by NukeManDan** [comment]
@xlc I will [scan the devhub](https://github.com/substrate-developer-hub/substrate-docs/issues/218) for any issues this may cause - anything else that should be done you think to ensure this is accounted for?

**Comment by xlc** [comment]
I am good as long as this is included in the release notes and we have docs about how to persistent data if people want to. I personally rely on this behavior because otherwise there could be lot of setup required to test some functionality.

**Review by gilescope — APPROVED** [review]

File: client/cli/src/commands/run_cmd.rs [hunk 1]
```diff
@@ -241,6 +241,8 @@ pub struct RunCmd {
 	///
 	/// Note: the directory is random per process execution. This directory is used as base path
 	/// which includes: database, node key and keystore.
+	///
+	/// When `--dev` is given and no explicit `--base-path`, this is option is implied.
```
Comment (apopiak) [inline client/cli/src/commands/run_cmd.rs:245]:
```suggestion
	/// When `--dev` is given and no explicit `--base-path`, this option is implied.
```

**Pull request merged**